mgr/dashboard: add flag to automatically deploy loki/promtail service at bootstrap

aaSharma14 opened the pull request:
Fixes: https://tracker.ceph.com/issues/56964
Signed-off-by: Aashish Sharma <aasharma@redhat.com>





<!--
  - Please give your pull request a title like

      [component]: [short description]

  - Please use this format for each git commit message:

      [component]: [short description]

      [A longer multiline description]

      Fixes: [ticket URL on tracker.ceph.com, create one if necessary]
      Signed-off-by: [Your Name] <[your email]>

    For examples, use "git log".
-->

## Contribution Guidelines
- To sign and title your commits, please refer to [Submitting Patches to Ceph](https://github.com/ceph/ceph/blob/main/SubmittingPatches.rst).

- If you are submitting a fix for a stable branch (e.g. "pacific"), please refer to [Submitting Patches to Ceph - Backports](https://github.com/ceph/ceph/blob/master/SubmittingPatches-backports.rst) for the proper workflow.

## Checklist
- Tracker (select at least one)
  - [ ] References tracker ticket
  - [ ] Very recent bug; references commit where it was introduced
  - [ ] New feature (ticket optional)
  - [ ] Doc update (no ticket needed)
  - [ ] Code cleanup (no ticket needed)
- Component impact
  - [ ] Affects [Dashboard](https://tracker.ceph.com/projects/dashboard/issues/new), opened tracker ticket
  - [ ] Affects [Orchestrator](https://tracker.ceph.com/projects/orchestrator/issues/new), opened tracker ticket
  - [ ] No impact that needs to be tracked
- Documentation (select at least one)
  - [ ] Updates relevant documentation
  - [ ] No doc update is appropriate
- Tests (select at least one)
  - [ ] Includes [unit test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/tests-unit-tests/)
  - [ ] Includes [integration test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/testing_integration_tests/)
  - [ ] Includes bug reproducer
  - [ ] No tests

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard cephadm`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`
- `jenkins test windows`
</details>

nizamial09 commented:
jenkins test dashboard cephadm

nizamial09 commented:
jenkins test dashboard cephadm

nizamial09 commented:
@adk3798 on the latest dashboard cephadm e2e job i saw that the grafana tests were failing. I checked locally and I saw that the grafana is unstable (alternating between `starting` and `running` state). 

[scrnli_29_07_2022_10-29-50.webm](https://user-images.githubusercontent.com/71764184/181703824-73b77ee7-1c61-4f5b-8c09-36d4912cdfb9.webm)

You can find the failing job here: https://jenkins.ceph.com/job/ceph-dashboard-cephadm-e2e-nightly-main/75/
and the mgr log for the same is here: https://jenkins.ceph.com/job/ceph-dashboard-cephadm-e2e-nightly-main/75/artifact/logs/mgr.cephadm.log

aaSharma14 commented:
Also modified the behaviour of deploying the promtail and loki services by default on bootstrap and changed the **--skip-centralized- logging** flag to **-add-centralized-logging** flag so if the user specifies this flag then only these two services are deployed on bootstrap.

adk3798 commented:
@aaSharma14 @nizamial09 I get that currently the dependency stuff is causing an issue, but in the long run is it correct for there to be a dependency here (i.e. should promtail and grafana be updating their config based on loki placement if it changes). The author who introduced the bug already created a patch to fix the dependency issue without removing it https://github.com/ceph/ceph/pull/47299 so we should decide here what the _correct_ behavior is. I think both patches would resolve the promtail and grafana daemons flapping.

@hookak if you're around, you may be interested in this

hookak commented:
First of all, I'm sorry for causing this issue.
I also found promtail and grafana flapping because of loki dependency, patch here https://github.com/ceph/ceph/pull/47299.
As @adk3798 commented, I agree that loki dependency is needed.

What do you guys think? @aaSharma14 @nizamial09

aaSharma14 commented:
If #47299 fixes the flapping issue, am good with keeping the loki dependency. Thanks for the fix @hookak

aaSharma14 commented:
@hookak @adk3798 , As discussed in the Dashboard daily standup, Is it fine if we don't deploy loki/promtail at bootstrap and instead of a **--skip-centralized-logging** flag, we provide a **--add-centralized-logging** flag to do the same. The reason is to avoid showing too many(relatively new services for the user) at the very beginning.

aaSharma14 commented:
jenkins test make check

nizamial09 commented on src/cephadm/cephadm:
```diff
@@ -9461,9 +9461,9 @@ def _get_parser():
         help='Do not automatically provision monitoring stack (prometheus, grafana, alertmanager, node-exporter)')
     parser_bootstrap.add_argument(
-        '--skip-centralized-logging',
+        '--add-centralized-logging',
```
this sounds better to me but if everyone's okay with `add` then I am good

```suggestion
        '--with-centralized-logging',
```

aaSharma14 replied:
Done.

adk3798 commented:
https://pulpito.ceph.com/adking-2022-08-03_01:34:50-orch:cephadm-wip-adk-testing-2022-08-02-1725-distro-default-smithi/

a handful of infra/teuthology issues. all passed in rerun https://pulpito.ceph.com/adking-2022-08-03_12:58:27-orch:cephadm-wip-adk-testing-2022-08-02-1725-distro-default-smithi/

adk3798 commented:
jenkins test api

adk3798 commented:
jenkins test windows

adk3798 commented:
https://pulpito.ceph.com/adking-2022-08-09_15:27:00-orch:cephadm-wip-adk-testing-2022-08-09-0854-distro-default-smithi/

failures due to infra issues with rhel tests. All passed in rerun
https://pulpito.ceph.com/adking-2022-08-09_20:24:21-orch:cephadm-wip-adk-testing-2022-08-09-0854-distro-default-smithi/

adk3798 commented:
jenkins test api

nizamial09 commented:
@adk3798 i went ahead and merged this one!
